ProcessLib/HT/HTFEM.h

```diff
@@ -110,6 +110,10 @@ class LocalAssemblerData : public HTLocalAssemblerInterface
             Eigen::Map<const Eigen::VectorXd>(&local_x[num_nodes], num_nodes);
 
         auto const & b = _process_data.specific_body_force.head(GlobalDim);
+        Eigen::MatrixXd unit_mat(GlobalDim, GlobalDim);
+        unit_mat.setZero();
+        for (std::size_t k(0); k < GlobalDim; ++k)
+            unit_mat(k,k) = 1.0;
```

Comment:
Actually eigen does provide the identity; have a look on the http://eigen.tuxfamily.org/dox/group__QuickRefPage.html page and search for  Identity there.